private/bufnew/bufconfig/generate_plugin_config.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufconfig
+
+// TODO: should this type live here? Perhaps it should live in the package that handles generation?
+// GenerateStrategy is the generation strategy for a protoc plugin.
+type GenerateStrategy int
+
+const (
+	// GenerateStrategyDirectory is the strategy to generate per directory.
+	//
+	// This is the default value for local plugins.
+	GenerateStrategyDirectory GenerateStrategy = 1
+	// GenerateStrategyAll is the strategy to generate with all files at once.
+	//
+	// This is the only strategy for remote plugins.
+	GenerateStrategyAll GenerateStrategy = 2
+)
+
+// GeneratePluginConfig is a configuration for a plugin.
+type GeneratePluginConfig interface {
```

Comment:
Are you asking callers to do a type switch to figure out what type of interface they are using?